src/main/java/com/barterbay/app/servcie/GoodService.java

```diff
@@ -0,0 +1,50 @@
+package com.barterbay.app.servcie;
+
+import com.barterbay.app.domain.Good;
+import com.barterbay.app.domain.dto.good.GoodCreatedDTO;
+import com.barterbay.app.mapper.AddressMapper;
+import com.barterbay.app.mapper.GoodMapper;
+import com.barterbay.app.repository.GoodRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+@AllArgsConstructor
+public class GoodService {
+
+  private final GoodRepository goodRepository;
+  private final GoodMapper goodMapper;
+  private final AddressMapper addressMapper;
+
+  @Transactional(readOnly = true)
+  public GoodCreatedDTO findGoodById(long goodId) {
+    return goodMapper.goodEntityToGoodDTO(goodRepository.findById(goodId).orElseThrow());
+  }
+
+  public void updateGoodById(long goodId, GoodCreatedDTO goodCreatedDTO) {
+    final Good goodById = goodRepository.findById(goodId).orElseThrow();
+    if (goodCreatedDTO.getGoodName() != null) {
```

Comment:
all this if statements should be in the mapper, you are supposed to create a method in mapper which will do all checks and mappings
Your service SHOULD NOT do any mappings - for this we use MupStruct, move all mapping  logic to your mapper